Studies/source/MvaTraining.cxx

```diff
@@ -372,45 +360,52 @@ class MVATraining {
         std::map<std::string, std::string> methods;
         for(const auto& point : grid){
             methods[options.GetName(point)+"_"+std::to_string(args.seed())] = options.GetConfigString(point);
-            std::cout << options.GetName(point)+"_"+std::to_string(args.seed()) << "    " << options.GetConfigString(point) << std::endl;
         }
         std::cout << methods.size() << " metodi" << std::endl;
 
-        for(const auto& m : methods)
-            factory->BookMethod(vars->loader.get(), TMVA::Types::kBDT, m.first, m.second);
-
-        TrainAllMethods(factory);
-//        factory->TrainAllMethods();
-        factory->TestAllMethods();
-        factory->EvaluateAllMethods();
+        std::map<std::string, double> ROCintegral;
+        std::map<std::string, std::vector<std::pair<std::string, double>>> importance;
 
         for(const auto& m : methods){
-            TString weightfile = ("mydataloader/weights/myFactory_"+m.first+".weights.xml");
-            vars->reader->BookMVA(m.first, weightfile);
+            auto factory = new TMVA::Factory ("myFactory", outfile.get(),"!V:!Silent:Color:DrawProgressBar:Transformations=I:AnalysisType=Classification");
```

Comment:
Use shared_ptr, otherwise factory will not be destroyed => memory will not be freed